addons/cert.py

```diff
@@ -234,6 +245,101 @@ def msc30(data):
         if simpleMatch(token, "rand ( )") and isStandardFunction(token):
             reportError(token, 'style', 'Do not use the rand() function for generating pseudorandom numbers', 'MSC30-c')
 
+# STR03-C
+# Do not inadvertently truncate a string
+def str03(data):
+    for token in data.tokenlist:
+        if token.str=='strncpy':
```

Comment:
It would feel better if you checked that there is a function call. You could at least use:

    simpleMatch(token, "strncpy (")